src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -348,13 +331,17 @@ private void StopActivity(HttpContext httpContext, Activity activity, bool hasDi
     }
 
     // These are versions of DiagnosticSource.Start/StopActivity that don't allocate strings per call (see https://github.com/dotnet/corefx/issues/37055)
+    [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(HttpContext))]
```

Comment:
FYI this will preserve public properties on HttpContext. It doesn't preserve all the types beneath HttpContext, e.g. HttpRequest. Or types on HttpRequest, e.g. Uri. etc

As far as I know, there isn't a good answer for diagnostic listener and trimming.

Reply:
Good point, is that why you explicitly wrote a ToString for the activity classes?

Reply:
I copied and pasted that. I'm guessing the purpose is for debugging, but I don't know for sure.

IMO DynamicDependency should have a recursive flag on it. Apply this policy to me, and everything on my public API.

Reply:
> IMO DynamicDependency should have a recursive flag on it. Apply this policy to me, and everything on my public API.

Doesn't work because of abstractions.

Reply:
Good point